client/src/components/FilesDialog/FilesDialog.vue

```diff
@@ -50,324 +367,7 @@
                 :disabled="(fileMode && !hasValue) || isBusy || (!fileMode && urlTracker.atRoot())"
                 @click="fileMode ? finalize() : selectLeaf(currentDirectory)">
```

Comment:
Ahh, thanks! I missed that one! Looks like typescript errors in .vue files don't get reported when compiling the client, I guess because they are not really typescript files... :thinking:

Reply:
Hmmm looking at that line it seems definitely fishy `selectLeaf` is meant for files, but `currentDirectory` is passed? I'll investigate what was this trying to do.

Reply:
Ok, I think this `selectLeaf` means really "select a single file or directory (no recursion)" I've refactored the name and moved the inline button click to a proper function which will show proper typescript errors when compiling if there are any :+1: